Mathlib/Data/Nat/Init.lean

```diff
@@ -434,6 +434,44 @@ protected lemma div_left_inj (hda : d ∣ a) (hdb : d ∣ b) : a / d = b / d ↔
   refine ⟨fun h ↦ ?_, congrArg fun b ↦ b / d⟩
   rw [← Nat.mul_div_cancel' hda, ← Nat.mul_div_cancel' hdb, h]
 
+/-- TODO: Replace `Nat.sub_mul_div` in core. -/
+lemma sub_mul_div' : (a - b * c) / b = a / b - c  := by
+  by_cases h : b * c ≤ a
+  · rw [Nat.sub_mul_div _ _ _ h]
+  · simp only [Nat.not_le] at h
+    rw [ Nat.sub_eq_zero_of_le (Nat.le_of_lt h),Nat.zero_div]
+    by_cases hn : b = 0
+    · simp only [hn, Nat.div_zero, zero_le, Nat.sub_eq_zero_of_le]
+    · have h2 : a / b ≤ (b * c) / b := Nat.div_le_div_right (Nat.le_of_lt h)
+      rw [Nat.mul_div_cancel_left _ (zero_lt_of_ne_zero hn)] at h2
+      rw [Nat.sub_eq_zero_of_le h2]
+
+lemma mul_sub_div_of_dvd (h1 : 0 < c) (h2 : c ∣ b) :  (c * a - b) / c = a - b / c := by
+  obtain ⟨_,hx⟩ := h2
+  simp only [hx,←Nat.mul_sub_left_distrib,Nat.mul_div_right,h1]
+
+lemma mul_add_mul_div_of_dvd  (hb : 0 < b) (hd : 0 < d) (h1 : b ∣ a) (h2 : d ∣ c) :
+    (a * d + b * c) / (b * d) = a / b + c / d := by
+  obtain ⟨n,hn⟩ := h1
+  obtain ⟨_,hm⟩ := h2
+  rw [hn,hm,Nat.mul_assoc b n d,Nat.mul_comm n d,← Nat.mul_assoc,← Nat.mul_assoc,
```

Comment:
Can you make sure your code is properly formatted? There are currently plenty of issues with it, eg:
```suggestion
  obtain ⟨n, hn⟩ := h1
  obtain ⟨_, hm⟩ := h2
  rw [hn, hm, Nat.mul_assoc b n d, Nat.mul_comm n d, ← Nat.mul_assoc, ← Nat.mul_assoc,
```

Reply:
I added the corresponding spaces after commas.